Xamarin.Forms.Platform.iOS/CollectionView/SelectableItemsViewRenderer.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.ComponentModel;
+
+namespace Xamarin.Forms.Platform.iOS
+{
+	public class SelectableItemsViewRenderer : ItemsViewRenderer
+	{
+		SelectableItemsView SelectableItemsView => (SelectableItemsView)Element;
+		SelectableItemsViewController SelectableItemsViewController => (SelectableItemsViewController)ItemsViewController;
+
+		protected override ItemsViewController CreateController(ItemsView itemsView, ItemsViewLayout layout)
+		{
+			return new SelectableItemsViewController(itemsView as SelectableItemsView, layout);
+		}
+
+		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs changedProperty)
+		{
+			base.OnElementPropertyChanged(sender, changedProperty);
+
+			if (changedProperty.Is(SelectableItemsView.SelectedItemProperty))
+			{
+				UpdateNativeSelection();
+			}
+			else if (changedProperty.Is(SelectableItemsView.SelectionModeProperty))
+			{
+				SelectableItemsViewController.UpdateSelectionMode();
+			}
+		}
+
+		protected override void SetUpNewElement(ItemsView newElement)
+		{
+			if (newElement != null && !(newElement is SelectableItemsView))
+			{
+				throw new ArgumentException($"{nameof(newElement)} must be of type {typeof(SelectableItemsView).Name}");
+			}
+
+			base.SetUpNewElement(newElement);
+
+			SelectableItemsViewController.UpdateSelectionMode();
+			UpdateNativeSelection();
+		}
+
+		void UpdateNativeSelection()
+		{
+			if (SelectableItemsView == null)
+			{
+				return;
+			}
+
+			var mode = SelectableItemsView.SelectionMode;
+			var selectedItem = SelectableItemsView.SelectedItem;
+
+			if (selectedItem == null)
+			{
+				if (mode == SelectionMode.None || mode == SelectionMode.Single)
+				{
+					ClearSelection();
+				}
+
+				// If the mode is Multiple and SelectedItem is set to null, don't do anything
+				return;
+			}
+
+			if (mode != SelectionMode.Multiple)
+			{
+				ClearSelection();
+				SelectableItemsViewController.SelectItem(selectedItem);
+			}
```

Comment:
This code looks iffy... we do a check for selected item == null, then inside that, we check for none/single. and then return.
Over here, we do a check for !multiple - which is the same as none || single.

They look to be the same type of action, but need confirmation as to the desired actions.

You might be wanting:

```csharp
if (mode != SelectionMode.Multiple) {
    ClearSelection();
    if (selectedItem != null) {
        SelectableItemsViewController.SelectItem(selectedItem);
    }
} else {
    // handle multiple selection...
}
```

Reply:
>  but need confirmation as to the desired actions

What needs confirmation?

Reply:
Simplified.